protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,494 @@
+     * @param epochId Epoch to claim for
+     */
+    function claim(uint256 epochId) external virtual override {
+        /// @notice cannot claim for current live epoch
+        EpochInfo storage info = epochs[epochId];
+        if (info.startTime == 0) { revert InvalidEpoch(); }
+        if (!info.hasEnded()) { revert CannotClaim(epochId); }
+
+        if (claimed[msg.sender][epochId]) { revert AlreadyClaimed(); }
+        uint256 bidTokenAmount = depositors[msg.sender][epochId];
+        if (bidTokenAmount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        claimed[msg.sender][epochId] = true;
+        SpiceAuctionConfig storage config = auctionConfigs[epochId];
+        (, address auctionToken) = _getBidAndAuctionTokens(config);
+
+        uint256 claimAmount = bidTokenAmount.mulDivRound(info.totalAuctionTokenAmount, info.totalBidTokenAmount, false);
+        claimedAmount[msg.sender][epochId] = claimAmount;
+        /// checkpoint claim for auction token
+        _claimedAuctionTokens[auctionToken] += claimAmount;
+        IERC20(auctionToken).safeTransfer(msg.sender, claimAmount);
+        emit Claim(msg.sender, epochId, bidTokenAmount, claimAmount);
+    }
+
+    /**
+     * @notice Check total bid token amount for epoch auction
+     * @param epochId Epoch to check for
+     */
+    function getAuctionTokenAmount(uint256 epochId) external override view returns (uint256) {
+        EpochInfo storage info = epochs[epochId];
+        return info.totalAuctionTokenAmount;
+    }
+
+    /**
+     * @notice Get total bid token amount for epoch auction
+     * @param epochId Epoch to get for
+     */
+    function getAuctionBidAmount(uint256 epochId) external override view returns (uint256) {
+        EpochInfo storage info = epochs[epochId];
+        return info.totalBidTokenAmount;
+    }
+
+    /**
+     * @notice Recover auction tokens for last but not started auction
+     * @param token Token to recover
+     * @param to Recipient
+     * @param amount Amount to auction tokens
+     */
+    function recoverToken(
+        address token,
+        address to,
+        uint256 amount
+    ) external override onlyDAOExecutor {
+        if (to == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        if (amount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+
+        if (token != spiceToken && token != templeGold) {
+            emit CommonEventsAndErrors.TokenRecovered(to, token, amount);
+            IERC20(token).safeTransfer(to, amount);
+            return;
+        }
+
+        uint256 epochId = _currentEpochId;
+        EpochInfo storage info = epochs[epochId];
+        /// @dev use `removeAuctionConfig` for case where `auctionStart` is called and cooldown is still pending
+        if (epochId != 0) {
+            if (info.startTime == 0) { revert InvalidConfigOperation(); }
+            if (!info.hasEnded() && auctionConfigs[epochId+1].duration == 0) { revert RemoveAuctionConfig(); }
```

Comment:
I don't understand this requirement. This effectively let you only recover tokens if either:
- the current epoch has ended
- configs for next auction have been set (duration > 0)

Why are those two conditions necessary for recovering tokens?